src/createAction.js

```diff
@@ -51,7 +51,7 @@ var createAction = function(definition) {
     }, Reflux.PublisherMethods, Reflux.ActionMethods, definition);
 
     var functor = function() {
-        functor[functor.sync?"trigger":"triggerAsync"].apply(functor, arguments);
+        return functor[functor.sync?"trigger":"triggerPromise"].apply(functor, arguments);
```

Comment:
This is the primary way `triggerAsync` is fired, you may want to deprecate `triggerAsync` or have it always call `triggerPromise`.  If you do that, you'll need to move `triggerAsync`'s logic into `triggerPromise`.